lib/srv/desktop/windows_server.go

```diff
@@ -1177,24 +1196,34 @@ func timer() func() int64 {
 func (s *WindowsService) generateUserCert(ctx context.Context, username string, ttl time.Duration, desktop types.WindowsDesktop, createUsers bool, groups []string) (certDER, keyDER []byte, err error) {
 	var activeDirectorySID string
 	if !desktop.NonAD() {
-		tc, err := s.tlsConfigForLDAP()
-		if err != nil {
-			return nil, nil, trace.Wrap(err)
-		}
+		// Use FnCache to fetch the SID, or load it from cache if we already have it
+		// The cache key is the username and domain combined to handle multi-domain setups
+		cacheKey := fmt.Sprintf("%s@%s", username, desktop.GetDomain())
+		sid, err := utils.FnCacheGet(ctx, s.sidCache, cacheKey, func(ctx context.Context) (string, error) {
+			tc, err := s.tlsConfigForLDAP()
```

Comment:
Now I wonder if we should cache the TLS config separately. Each time we do this we have to first connect to auth and have it issue a certificate before we can connect to LDAP.

If we issued certificates that were valid for say 8 hours, and only cached them for 4 hours, that would require a lot fewer certificates to be issued, and we should be safe from using expired certs since they expire from the cache long before they actually expire.

Thoughts?

(Note: if we do decide to do this, I don't think we need `FnCache` for this case since there's only ever one TLS config at most in the cache.)